File: spring-web-java/src/main/java/spring/web/java/config/security/UserDetailsServiceImpl.java
package spring.web.java.config.security;

import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import spring.web.java.repository.MemberRepository;

@RequiredArgsConstructor
public class UserDetailsServiceImpl implements UserDetailsService {

    private final MemberRepository memberRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        return memberRepository.findByAccount(username)
                .map(UserDetailsImpl::new)
                .orElseThrow(() -> new UsernameNotFoundException("유저 없음"));
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/config/security/UserDetailsServiceImpl.kt
package spring.web.kotlin.config.security

import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import spring.web.kotlin.repository.MemberRepository

class UserDetailsServiceImpl(
    private val memberRepository: MemberRepository
) : UserDetailsService {
    override fun loadUserByUsername(username: String) =
        memberRepository.findByAccount(username).orElse(null)
            ?.let { UserDetailsImpl(it) }
            ?: throw UsernameNotFoundException("유저 없음")
}